.github/actions/multi-approvers/src/multi-approvers.ts

```diff
@@ -0,0 +1,307 @@
+
+type PullRequestReview =
+  RestEndpointMethodTypes["pulls"]["listReviews"]["response"]["data"];
+type Octokit = ReturnType<typeof getOctokit>;
+export type EventName = "pull_request" | "pull_request_review";
+
+export function isEventName(v: string): v is EventName {
+  return ["pull_request", "pull_request_review"].includes(v);
+}
+
+const MIN_APPROVED_COUNT = 2;
+
+export interface MultiApproversParams {
+  eventName: EventName;
+  runId: number;
+  branch: string;
+  pullNumber: number;
+  repoName: string;
+  repoOwner: string;
+  token: string;
+  team: string;
+  octokitOptions?: OctokitOptions;
+  // eslint-disable-next-line no-unused-vars
+  logDebug: (_: string) => void;
+  // eslint-disable-next-line no-unused-vars
+  logInfo: (_: string) => void;
+  // eslint-disable-next-line no-unused-vars
+  logNotice: (_: string) => void;
+}
+
+export class MultiApproversAction {
+  private readonly isInternalCache = new Map<string, boolean>();
+  private readonly eventName: string;
+  private readonly runId: number;
+  private readonly branch: string;
+  private readonly pullNumber: number;
+  private readonly repoName: string;
+  private readonly repoOwner: string;
+  private readonly team: string;
+  private readonly octokit: Octokit;
+
+  constructor(params: MultiApproversParams) {
+    this.eventName = params.eventName;
+    this.runId = params.runId;
+    this.branch = params.branch;
+    this.pullNumber = params.pullNumber;
+    this.repoName = params.repoName;
+    this.repoOwner = params.repoOwner;
+    this.team = params.team;
+    this.logDebug = params.logDebug;
+    this.logInfo = params.logInfo;
+    this.logNotice = params.logNotice;
+
+    this.octokit = getOctokit(params.token, params.octokitOptions);
+  }
+
+  // Set in the constructor.
+  // eslint-disable-next-line no-unused-vars
+  private logDebug: (_: string) => void;
+
+  // Set in the constructor.
+  // eslint-disable-next-line no-unused-vars
+  private logInfo: (_: string) => void;
+
+  // Set in the constructor.
+  // eslint-disable-next-line no-unused-vars
+  private logNotice: (_: string) => void;
+
+  // TODO(ishafer): Consider using a memoization library.
+  private async isInternal(login: string): Promise<boolean> {
+    const cachedValue = this.isInternalCache.get(login);
+    if (cachedValue !== undefined) {
+      return cachedValue;
+    }
+    const value = await this._isInternal(login);
+    this.isInternalCache.set(login, value);
+    return value;
+  }
+
+  // Tests whether the given login is an active member of the team.
+  private async _isInternal(login: string): Promise<boolean> {
+    try {
+      const response = await this.octokit.rest.teams.getMembershipForUserInOrg({
+        org: this.repoOwner,
+        team_slug: this.team,
+        username: login,
+      });
+      const state = response.data.state;
+      if (state !== "active") {
+        this.logDebug(`Skipping because state is not active: ${state}`);
```

Comment:
I should have caught this earlier, but we probably want to log the username too

Reply:
Done. And added to a couple other places in the same method.